src/http_microhttpd.c

```diff
@@ -589,12 +593,20 @@ static int encode_and_redirect_to_splashpage(struct MHD_Connection *connection,
 		}
 	}
 
-	if (encoded[0]) {
+	if (config->fas_port) {
+		target = (config->fas_remoteip ? config->fas_remoteip : config->gw_address);
+
+		// append clientip or authaction url
+		if (config->fas_secure_enabled) {
+			safe_asprintf(&splashpageurl, "http://%s:%u/%s?clientip=%s&redir=%s",
+				target, config->fas_port, config->fas_path, token, encoded);
```

Comment:
clientip=<client_token>
This looks like a hack or a bug. Shouldn't it be tok=<client_token>?

Reply:
Same hack as mentioned above ;-)
Fix on its way.